vercel/next.js
# mknichel/backport multi zones docs

- use pathToFileUrl to make esm import()s work with absolute windows paths (#64386)
- update publish configs
- modify scripts
- v14.2.1
- Fix Server Action error logs for unhandled POST requests (#64315)
- Improve rendering performance (#64408)
- Fix the method prop case in Server Actions transform (#64398)
- fix(next-lint): update option --report-unused-disable-directives to --report-unused-disable-directives-severity (#64405)
- tweak test for Azure (#64424)
- router restore should take priority over pending actions (#64449)
- Fix client boundary inheritance for barrel optimization (#64467)
- improve turborepo caching (#64493)
- feat: strip traceparent header from cachekey (#64499)
- Update font data (#64481)
- Fix more Turbopack build tests (#64384)
- Update lockfile for compatibility with turbo (#64360)
- Fix typo in dynamic-rendering.ts (#64365)
- Fix DynamicServerError not being thrown in fetch (#64511)
- fix(next): `Metadata.openGraph` values not resolving basic values when `type` is set (#63620)
- disable production chunking in dev (#64488)
- Fix cjs client components tree-shaking (#64558)
- fix refresh behavior for discarded actions (#64532)
- fix: filter out middleware requests in logging (#64549)
- Turbopack: Allow client components to be imported in app routes (#64520)
- Fix ASL bundling for dynamic css (#64451)
- add pathname normalizer for actions (#64592)
- fix incorrect refresh request when basePath is set (#64589)
- test: skip turbopack build test (#64356)
- hotfix(turbopack): Update with patch for postcss.config.js path resolution on Windows (#64677)
- v14.2.2
- Fix: resolve mixed re-exports module as cjs (#64681)
- fix: mixing namespace import and named import client components (#64809)
- Fix mixed exports in server component with barrel optimization (#64894)
- Fix next/image usage in mdx (#64875)
- fix(fetch-cache): fix additional typo, add type & data validation (#64799)
- prevent erroneous route interception during lazy fetch (#64692)
- fix root page revalidation when redirecting in a server action (#64730)
- fix: remove traceparent from cachekey should not remove traceparent from original object (#64727)
- Clean-up fetch metrics tracking (#64746)
- v14.2.3
- fix: ensure route handlers properly track dynamic access (#66446)
- lock CNA tests to 14.2.3
- fix another test
- fix NextRequest proxy in edge runtime (#66551)
- Update turbopack to turbopack-240605-14.x-hotfix.1
- Fix next/dynamic with babel and src dir (#65177)
- Use vercel deployment url for metadataBase fallbacks (#65089)
- fix(next/image): detect react@19 for `fetchPriority` prop (#65235)
- Fix loading navigation with metadata and prefetch (#66447)
- prevent duplicate RSC fetch when action redirects (#66620)
- ensure router cache updates reference the latest cache values (#66681)
- Prevent append of trailing slash in cases where path ends with a file extension (#66636)
- Fix inconsistency with 404 getStaticProps cache-control (#66674)
- Use addDependency to track metadata route file changes (#66714)
- Add timeout/retry handling for fetch cache (#66652)
- fix: app-router prefetch crash when an invalid URL is passed to Link (#66755)
- [not a backport] fix lint errors
- v14.2.4
- fix: Backport class properties fix (#67377)
- fix(next): initial prefetch cache not set properly with different search params (#65977)
- perf: improve retrieving `versionInfo` on Turbo HMR (#67309)
- fix(next/image): handle invalid url (#67465)
- Fix noindex is missing on static not-found page (#67135)
- Use addDependency to track metadata route file changes (#66714)
- fix(next-swc): correct path interop to filepath for wasm (#65633)
- fix(next): reject protocol-relative URLs in image optimization (#65752)
- Fix server action edge redirect with middleware rewrite (#67148)
- avoid merging global css in a way that leaks into other chunk groups (#67373)
- Upgrade acorn (#67592)
- Ensure turbo is setup when building in docker (#66804)
- Log stdio for pull-turbo-cache script (#66759)
- fix: metadata route merging conflicts
- v14.2.5
- [docs] Backport Multi-Zones docs to 14.x branch
